src/gt4sd/training_pipelines/__init__.py

```diff
@@ -208,16 +210,15 @@ def training_pipeline_name_to_metadata(name: str) -> Dict[str, Any]:
     Returns:
         dictionary describing the parameters of the pipeline. If the pipeline is not found, no metadata (a.k.a., an empty dictionary is returned).
     """
+    file_manager = ExitStack()
+    atexit.register(file_manager.close)
+
     metadata: Dict[str, Any] = {"training_pipeline": name, "parameters": {}}
     if name in TRAINING_PIPELINE_NAME_METADATA_MAPPING:
         try:
-            with open(
-                pkg_resources.resource_filename(
-                    "gt4sd",
-                    f"training_pipelines/{TRAINING_PIPELINE_NAME_METADATA_MAPPING[name]}",
-                ),
-                "rt",
-            ) as fp:
+            ref = importlib_resources.files("my.package") / "resource.dat"
```

Comment:
I really struggle to follow the logic here, maybe it's a problem of copy-pasting leftovers.

Reply:
I would consider also using here the method we create for usage in the tests.

Reply:
ah great catch, indeed a copy-paste leftover. Strange the tests passed anyways